lib/highLevelConsumer.js

```diff
@@ -171,45 +169,43 @@ HighLevelConsumer.prototype.connect = function () {
                             return p.topic;
                         });
                         self.client.refreshMetadata(topicNames, function (err) {
-                            register();
                             self.rebalancing = false;
                             if (err) {
-                              self.emit('error', err);
+                                self.emit('error', err);
                             } else {
-                              self.emit('rebalanced');
+                                self.emit('rebalanced');
+                            }
+                            if (self.pending) {
+                                self.pending = false;
+                                rebalance();
                             }
                         });
                     }
                 });
             });
+        } else {
+            self.pending = true;
         }
     }
 
-    // Wait for the consumer to be ready
-    this.on('registered', function () {
-        rebalance();
-    });
-
-    function register() {
+    function registerListeners() {
         debug("Registered listeners");
         // Register for re-balances (broker or consumer changes)
         self.client.zk.on('consumersChanged', rebalance);
         self.client.on('brokersChanged', rebalance);
     }
-
-    function deregister() {
-        debug("Deregistered listeners");
-        // Register for re-balances (broker or consumer changes)
-        self.client.zk.removeListener('consumersChanged', rebalance);
-        self.client.removeListener('brokersChanged', rebalance);
-    }
+    self.on('registered', function () {
+        setTimeout(function () {
+            registerListeners();
+            self.rebalance();
```

Comment:
fixed.

I exposed rebalance method to give outside programmer more control in kafka-node2, and simply copy2paste make this error, sorry.